plugins/vpp/ifplugin/vppcalls/dump_interface_vppcalls.go

```diff
@@ -455,11 +461,20 @@ func (handler *ifVppHandler) dumpDhcpClients() (map[uint32]*Dhcp, error) {
 			return nil, err
 		}
 		client := dhcpDetails.Client
+		lease := dhcpDetails.Lease
+
+		var hostMac net.HardwareAddr = lease.HostMac
 
 		dhcpData[client.SwIfIndex] = &Dhcp{
 			IfIdx:            client.SwIfIndex,
-			Hostname:         string(client.Hostname),
-			ID:               string(client.ID),
+			State:            lease.State,
+			Hostname:         string(bytes.SplitN(client.Hostname, []byte{0x00}, 2)[0]),
+			ID:               string(bytes.SplitN(client.ID, []byte{0x00}, 2)[0]),
+			HostAddress:      string(bytes.SplitN(lease.HostAddress, []byte{0x00}, 2)[0]),
```

Comment:
No cutting 0x00 bytes for addresses, just ASCII names.

Reply:
Also, these addresses should be changed into `net.IP` and to string.. since this is binary form of IP.